### ISSUE TITLE
Failed to install the pycurl as install_requires

### ISSUE THREAD

**Issue by iamtekson** [readthedocs/readthedocs.org#8184]
## Details

* Read the Docs project URL:  https://geoserver-rest.readthedocs.io/
* Build URL (if applicable): https://geoserver-rest.readthedocs.io/
* Read the Docs username (if applicable):  iamtekson

## Expected Result
It should build the documentation without any issue

*A description of what you wanted to happen*
I have already updated the docs for my `geoserver-rest` package. This time I just have a small updated on documentation. When I tried to build the docs, it throws the following error,

```bash
Collecting pycurl
  Downloading pycurl-7.43.0.6.tar.gz (222 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/docs/checkouts/readthedocs.org/user_builds/geoserver-rest/envs/latest/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-93q4v_7e/pycurl_0da21a0f5f524aa1a745ae15f342935a/setup.py'"'"'; __file__='"'"'/tmp/pip-install-93q4v_7e/pycurl_0da21a0f5f524aa1a745ae15f342935a/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-onn4qi5v
         cwd: /tmp/pip-install-93q4v_7e/pycurl_0da21a0f5f524aa1a745ae15f342935a/
    Complete output (22 lines):
    Traceback (most recent call last):
      File "/tmp/pip-install-93q4v_7e/pycurl_0da21a0f5f524aa1a745ae15f342935a/setup.py", line 237, in configure_unix
        stdout=subprocess.PIPE, stderr=subprocess.PIPE)
      File "/home/docs/.pyenv/versions/3.7.9/lib/python3.7/subprocess.py", line 800, in __init__
        restore_signals, start_new_session)
      File "/home/docs/.pyenv/versions/3.7.9/lib/python3.7/subprocess.py", line 1551, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'curl-config': 'curl-config'
```

**Comment by stsewd** [comment]
Hi, I'm getting the same error locally with `pip install pycurl`. I'm using fedora, we use ubuntu images to build your docs, so I guess there is a problem installing that package on linux. What OS are you using to install locally? but anyway, you should report this to https://github.com/pycurl/pycurl/issues

**Comment by iamtekson** [comment]
I am using ubuntu and windows OS. It is working fine on my local server. But previously, I was able to install it and generate the documentation. But right now I don't know what is the issue.

**Comment by iamtekson** [comment]
I found this issue in the pycurl repo: https://github.com/pycurl/pycurl/issues/676

can you suggest to me how to solve this issue in readthedocs?

**Comment by stsewd** [comment]
@iambenzo looks like the solution is to install some apt-packages

https://github.com/pycurl/pycurl/issues/596#issuecomment-557302996

That's going to be possible in a couple of hours :D https://docs.readthedocs.io/en/stable/config-file/v2.html#build-apt-packages (we are about to deploy that change today)

**Comment by iambenzo** [comment]
I don't think I was supposed to be mentioned here. Tagging @iamtekson for notification 👍

**Comment by stsewd** [comment]
@iamtekson the change is live, you should be able to use that feature now :)

**Comment by iamtekson** [comment]
@stsewd, I just tried but got still the same issue. Are you sure it is ok now?

**Comment by stsewd** [comment]
@iamtekson you need to create a config file with the packages you need to install https://github.com/gicait/geoserver-rest https://docs.readthedocs.io/en/stable/config-file/v2.html#build-apt-packages

**Comment by iamtekson** [comment]
Hi @stsewd, Sorry to disturb you again. But I got confused that where should I write the configuration and how to link with readthedocs? Can you please give me an example quickly? 

```shell
version: 2

build:
  apt_packages:
    - libssl-dev 
    - libcurl4-openssl-dev 
    - python3-dev
    - cmake
```

**Comment by stsewd** [comment]
@iamtekson you need to put that content in a `.readthedocs.yaml` file in your repository and rtd will read it from there, you can read the docs https://docs.readthedocs.io/en/stable/config-file/v2.html for more information on how to use a config file.

```yaml
version: 2

build:
  apt_packages:
    - libssl-dev
    - libcurl4-openssl-dev
    - python3-dev

sphinx:
   configuration: docs/source/conf.py

python:
   version: 3.8
   install:
   - requirements: requirements.txt
```

**Comment by iamtekson** [comment]
Perfect!
Thank you very much for the information.

**Comment by iamtekson** [comment]
Again I was stuck on the `gdal` installation issue. After doing some research on `readthedocs` structure, I got to know that  I don't required to install the `gdal` and `pyculr` for generating the docs. So I created the separated `requirements-docs.txt` file for docs as follows,

```txt
sphinx==2.4.3
sphinx_rtd_theme==0.4.3
readthedocs-sphinx-ext==1.0.1
m2r==0.2.1
mistune==0.8.4
sphinxcontrib-httpdomain==1.7.0
sphinxcontrib-openapi==0.7.0
```

And my `.readthedocs` file looks like below (I don't need to add the `build` section anymore),

```bash
version: 2

sphinx:
   configuration: docs/source/conf.py

python:
   install:
   - requirements: docs/requirements-docs.txt
```

The installation of apt packages as explained above is a pretty cool feature. It installs the package successfully. But I don't want to make it complex. That's why I removed the `build` section. Just want to mention [this PR](https://github.com/readthedocs/blog/pull/104)
